packages/block-editor/src/components/rich-text/embed-handler-picker.native.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useRef, useImperativeHandle } from '@wordpress/element';
+import { Picker } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const DEFAULT_PICKER_OPTIONS = [
+	{
+		id: 'createEmbed',
+		label: __( 'Create embed' ),
+		value: 'createEmbed',
+		onSelect: noop,
+	},
+	{
+		id: 'createLink',
+		label: __( 'Create link' ),
+		value: 'createLink',
+		onSelect: noop,
+	},
+];
+
+export default forwardRef( ( {}, ref ) => {
+	const pickerRef = useRef();
+	const pickerOptions = useRef( DEFAULT_PICKER_OPTIONS ).current;
+
+	function onPickerSelect( value ) {
+		const selectedItem = pickerOptions.find(
+			( item ) => item.value === value
+		);
+		selectedItem.onSelect();
+	}
```

Comment:
Good idea! Just to verify, once this is done, I would need to utilize some form of tracing mechanism to verify the number of re-renders taking place. What tool do you utilize to do this?

Reply:
@fluiddot I did this operation here be43155 579def2

Reply:
> Good idea! Just to verify, once this is done, I would need to utilize some form of tracing mechanism to verify the number of re-renders taking place. What tool do you utilize to do this?

I completely missed this comment, sorry. I usually measure renders by adding logs but I think you could use [Flipper](https://fbflipper.com/) and the [React devtools](https://fbflipper.com/docs/features/react-native/) that are already included for this.

Reply:
I did a quick test and found out that doesn't prevent re-renders of the `Picker` component because the parent is the one that forces the render. The only way I managed to prevent extra renders is by memoizing the component (see attached diff patch). Nevertheless, it's a good practice to wrap callbacks passed in props with `useCallback`, so let's stick with this change 👍 .

In the following screenshots, you can see that independently on using `useCallback` it renders the `Picker` component three times:

Without `useCallback`|With `useCallback`
--|--
<img width="825" alt="no-callback" src="https://user-images.githubusercontent.com/14905380/136034185-ad6ff30c-0e4f-482d-9ff3-59271b4a7980.png">|<img width="835" alt="with-callback" src="https://user-images.githubusercontent.com/14905380/136034221-c009c110-04f0-45e7-a001-ca2ba81374ef.png">

<details><summary>Click here to see the diff patch</summary>

```diff
diff --git a/packages/block-editor/src/components/rich-text/embed-handler-picker.native.js b/packages/block-editor/src/components/rich-text/embed-handler-picker.native.js
index 686e55b241a2aca54102fd5ec8974f585e0e8fa9..c4ffc0894e38a74b17effbe6710ae145c04329bd 100644
--- a/packages/block-editor/src/components/rich-text/embed-handler-picker.native.js
+++ b/packages/block-editor/src/components/rich-text/embed-handler-picker.native.js
@@ -10,6 +10,7 @@ import {
 	forwardRef,
 	useRef,
 	useImperativeHandle,
+	memo,
 	useCallback,
 } from '@wordpress/element';
 import { Picker } from '@wordpress/components';
@@ -30,7 +31,7 @@ const DEFAULT_PICKER_OPTIONS = [
 	},
 ];
 
-export default forwardRef( ( {}, ref ) => {
+const EmbedHandlerPicker = forwardRef( ( {}, ref ) => {
 	const pickerRef = useRef();
 	const pickerOptions = useRef( DEFAULT_PICKER_OPTIONS ).current;
 
@@ -62,3 +63,4 @@ export default forwardRef( ( {}, ref ) => {
 		/>
 	);
 } );
+export default memo( EmbedHandlerPicker );

````

</details>